CONTRIBUTIONS.adoc

```diff
@@ -0,0 +1,269 @@
+and we do not go through them in more detail. We do, however, recommend
+to create dedicated topic branches for pull requests instead of creating
+them based on the master branch. This is especially important if you
+plan to work on multiple pull requests at the same time.
+
+=== Coding conventions
+
+==== General guidelines
+
+Robot Framework uses the general Python code conventions defined in
+https://www.python.org/dev/peps/pep-0008/[PEP-8]. In addition to that,
+we try to write
+http://python.net/~goodger/projects/pycon/2007/idiomatic/handout.html[idiomatic
+Python] and follow the
+https://en.wikipedia.org/wiki/SOLID_(object-oriented_design)[SOLID
+principles] with all new code. An important guideline is that the code
+should be clear enough that comments are generally not needed.
+
+All code, including test code, must be compatible with all supported
+Python interpreters and versions. Most importantly this means that the
+code must support both Python 2 and Python 3.
+
+==== Line length
+
+Maximum line length with Python code, including docstrings and comments,
+is 88 characters. This is also what
+https://pypi.org/project/black/[Black] uses by default and
+https://black.readthedocs.io/en/stable/the_black_code_style.html#line-length[their
+documentation] explains why. Notice that we do not have immediate plans
+to actually take Black into use but we may consider that later.
+
+With Robot Framework tests the maximum line length is 100.
+
+==== Whitespace
+
+We are pretty picky about using whitespace. We follow link:[PEP-8] in
+how to use blank lines and whitespace in general, but we also have some
+stricter rules:
+
+* No blank lines inside functions.
+* No blank lines between a class declaration and class attributes or
+between attributes.
+* Indentation using spaces, not tabs.
+* No trailing spaces.
+* No extra empty lines at the end of the file.
+* Files must end with a newline.
+
+Most of these rules are such that any decent text editor or IDE can be
+configured to automatically format files according to them.
+
+==== Docstrings
+
+Docstrings should be added to public APIs, but they are not generally
+needed in internal code. When docstrings are added, they should follow
+https://www.python.org/dev/peps/pep-0257/[PEP-257]. See
+link:#api-documentation[API documentation] section below for more
+details about documentation syntax, generating API docs, etc.
+
+=== Documentation
+
+With new features adequate documentation is as important as the actual
+functionality. Different documentation is needed depending on the issue.
+
+==== User Guide
+
+Robot Framework's features are explained in the
+http://robotframework.org/robotframework/#user-guide[User Guide]. It is
+generated using a custom script based on the source in
+http://docutils.sourceforge.net/rst.html[reStructuredText] format. For
+more details about editing and generating it see
+doc/userguide/README.rst.
+
+==== Libraries
```

Comment:
This section must be properly adjusted to RIDE:
- we don't provide Libraries
- important to know how to run unit tests
- important to know how to run RIDE (with invoke)
- important to know how to run RIDE using the debug option